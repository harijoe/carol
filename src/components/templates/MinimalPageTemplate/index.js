import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'
import { addLocaleData } from 'react-intl'
import fr from 'react-intl/locale-data/fr'
import en from 'react-intl/locale-data/en'
import es from 'react-intl/locale-data/es'
import { injectGlobals } from 'utils/style'

import defaultTheme, { resets, scaffolding } from '../../themes/default'

injectGlobals([resets, scaffolding])

addLocaleData([...es, ...en, ...fr])

const MinimalPageTemplate = ({ children }) =>
  <ThemeProvider theme={defaultTheme}>
    {children}
  </ThemeProvider>

MinimalPageTemplate.propTypes = {
  children: PropTypes.any.isRequired,
}

export default MinimalPageTemplate
