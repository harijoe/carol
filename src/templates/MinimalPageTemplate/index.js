import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'
import defaultTheme from 'theme/default'

const MinimalPageTemplate = ({ children }) =>
  <ThemeProvider theme={defaultTheme}>
    {children}
  </ThemeProvider>

MinimalPageTemplate.propTypes = {
  children: PropTypes.any.isRequired,
}

export default MinimalPageTemplate
