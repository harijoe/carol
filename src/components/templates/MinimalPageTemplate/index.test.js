Object.defineProperty(window.location, 'hostname', {
  writable: true,
  value: 'carol-co-uk.dev.quotatis.net',
})

import React from 'react'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'

import MinimalPageTemplate from './'
import mockStore from '../../../../test/storeMock'

const store = mockStore()

const wrapper = shallow(
  <Provider store={store}>
    <MinimalPageTemplate header="header">test</MinimalPageTemplate>
  </Provider>
)

it('renders MinimalPageTemplate', () => {
  expect(wrapper).toMatchSnapshot()
})
