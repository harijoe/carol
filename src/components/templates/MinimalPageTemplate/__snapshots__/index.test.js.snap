// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MinimalPageTemplate 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MinimalPageTemplate
      header="header"
    >
      test
    </MinimalPageTemplate>
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "children": "test",
      "header": "header",
    },
    "ref": null,
    "rendered": "test",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": "test",
        "header": "header",
      },
      "ref": null,
      "rendered": "test",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
